File: main3.go
package main

import "fmt"

func countEvenDigits(num int) int {
    if num < 10 {
        return -1
    }
    
    count := 0
    temp := num
    
    for temp > 0 {
        digit := temp % 10
        if digit%2 == 0 {
            count++
        }
        temp /= 10
    }
    
    return count
}

func main() {
    var N int
    
    fmt.Print("Введите количество чисел N: ")
    fmt.Scan(&N)
    
    numbers := make([]int, N)
    
    fmt.Printf("Введите %d чисел: ", N)
    for i := 0; i < N; i++ {
        fmt.Scan(&numbers[i])
    }
    
    fmt.Print("Результат: ")
    
    for i := 0; i < N; i++ {
        result := countEvenDigits(numbers[i])
        
        if result == -1 {
            fmt.Print("-")
        } else {
            fmt.Print(result)
        }
        
        if i < N-1 {
            fmt.Print(" ")
        }
    }
    
    fmt.Println()
}

File: main3.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int countEvenDigits(int num) {
    if (num < 10) {
        return -1;
    }
    
    int count = 0;
    int temp = num;
    
    while (temp > 0) {
        int digit = temp % 10;
        if (digit % 2 == 0) {
            count++;
        }
        temp /= 10;
    }
    
    return count;
}

int main() {
    int N;
    
    cout << "Введите количество чисел N: ";
    cin >> N;
    
    vector<int> numbers(N);
    
    cout << "Введите " << N << " чисел: ";
    for (int i = 0; i < N; i++) {
        cin >> numbers[i];
    }
    
    cout << "Результат: ";
    
    for (int i = 0; i < N; i++) {
        int result = countEvenDigits(numbers[i]);
        
        if (result == -1) {
            cout << "-";
        } else {
            cout << result;
        }
        
        if (i < N - 1) {
            cout << " ";
        }
    }
    
    cout << endl;
    
    return 0;
}